lib/reporting/reports/sales_tax/sales_tax_totals_by_order.rb

```diff
@@ -159,7 +159,10 @@ def non_adjusted_tax_rate_total(query_result_row)
         end
 
         def voucher_tax_adjustment(order)
-          order.all_adjustments.voucher_tax.first&.amount || 0
+          included_tax_voucher_adjustment = order.voucher_adjustments.first&.included_tax || 0
```

Comment:
I also found this part hard to understand. Eg `Adjustment` also has method `included_tax_total` which looks up any associated tax adjustments. It seems to be a complex web.. and probably I'm just not following it today.

Reply:
`included_tax_total` uses adjustments with `originator_type` of `Spree::TaxRate` as you noted. In case of a voucher adjustment we store tax data in 2 different ways: 
- tax included in price, tax data is stored in `included_tax` in the voucher adjustment
- tax excluded in price, tax data is stored in it's own `voucher_adjustment` with a label that starts with "Tax", which is a bit clunky